packages/supi/src/install/index.ts

```diff
@@ -776,7 +797,16 @@ const _installInContext: InstallFunction = async (projects, ctx, opts) => {
   }
 
   const lockfileOpts = { forceSharedFormat: opts.forceSharedLockfile }
-  if (!opts.lockfileOnly && opts.enableModulesDir) {
+  if (opts.fixLockfile) {
```

Comment:
the --fix-lockfile works like --lockfile-only, only regenerate missing lockfiles fields

Reply:
can't we just set lockfileOnly to true when fixLockfile is true?

Reply:
We can't. because just set lockfileOnly to true won't fix requiresBuild(because requiresBuild always read existing lockfile for performance, and we can't change this behavior as we talked in another comment), but here we can fix all fields. I think fixLockfile should work as its name, fixing all fields.

this is the most lightweight way I can figure out to achieve this feature